source/state_representation/include/state_representation/trajectory/TrajectoryBase.hpp

```diff
@@ -0,0 +1,240 @@
+inline TrajectoryBase<TrajectoryT>::TrajectoryBase(const std::string& name) : State(name) {
+  this->reset();
+}
+
+template<typename TrajectoryT>
+inline void TrajectoryBase<TrajectoryT>::reset() {
+  this->State::reset();
+  this->points_.clear();
+}
+
+template<typename TrajectoryT>
+inline void TrajectoryBase<TrajectoryT>::add_point(const TrajectoryT& new_point) {
+  this->set_empty(false);
+  this->points_.push_back(new_point);
+}
+
+template<typename TrajectoryT>
+inline void TrajectoryBase<TrajectoryT>::insert_point(const TrajectoryT& new_point, unsigned int pos) {
+  if (pos > this->points_.size()) {
+    throw std::out_of_range("Index out of range");
+  }
+  this->set_empty(false);
+  auto it_points = this->points_.begin();
+  std::advance(it_points, pos);
+  this->points_.insert(it_points, new_point);
+}
+
+template<typename TrajectoryT>
+inline void TrajectoryBase<TrajectoryT>::delete_point() {
+  this->set_empty(false);
+  if (!this->points_.empty()) {
+    this->points_.pop_back();
+  }
+}
+
+template<typename TrajectoryT>
+inline void TrajectoryBase<TrajectoryT>::clear() {
+  this->points_.clear();
+}
+
+template<typename TrajectoryT>
+inline const std::deque<TrajectoryT>& TrajectoryBase<TrajectoryT>::get_points() const {
+  return this->points_;
+}
+
+template<typename TrajectoryT>
+inline const TrajectoryT& TrajectoryBase<TrajectoryT>::get_point(unsigned int index) const {
+  return this->points_[index];
+}
+
+template<typename TrajectoryT>
+inline TrajectoryT& TrajectoryBase<TrajectoryT>::get_point(unsigned int index) {
+  return this->points_[index];
+}
+
+template<typename TrajectoryT>
+inline void TrajectoryBase<TrajectoryT>::set_point(const TrajectoryT& point, unsigned int index) {
+  if (index >= this->points_.size()) {
+    throw std::out_of_range("Index out of range");
+  }
+  this->points_[index] = point;
+}
+
+template<typename TrajectoryT>
+inline void TrajectoryBase<TrajectoryT>::set_points(const std::vector<TrajectoryT>& points) {
+  if (points.size() == 0) {
+    return;
+  }
+  this->clear();
+  for (unsigned int i = 0; i < points.size(); ++i) {
+    this->add_point(points[i]);
+  }
+}
+
+template<typename TrajectoryT>
+inline const std::deque<std::chrono::nanoseconds> TrajectoryBase<TrajectoryT>::get_durations() const {
+  std::deque<std::chrono::nanoseconds> durations;
+  for (unsigned int i = 0; i < this->points_.size(); ++i) {
+    durations.push_back(this->points_[i].duration);
+  }
+  return durations;
+}
+
+template<typename TrajectoryT>
+inline const std::deque<std::chrono::nanoseconds> TrajectoryBase<TrajectoryT>::get_times_from_start() const {
+  std::deque<std::chrono::nanoseconds> times_from_start;
+  for (unsigned int i = 0; i < this->points_.size(); ++i) {
+    std::chrono::nanoseconds time_from_start = std::chrono::nanoseconds(0);
+    for (unsigned int j = 0; j <= i; ++j) {
+      time_from_start += this->points_[j].duration;
+    }
+    times_from_start.push_back(time_from_start);
+  }
```

Comment:
```suggestion
  std::chrono::nanoseconds time_from_start = std::chrono::nanoseconds(0);
  for (unsigned int i = 0; i < this->points_.size(); ++i) {
    time_from_start += this->points_[i].duration;
    times_from_start.push_back(time_from_start);
  }
```
or am I missing something?